vyper/compiler/phases.py

```diff
@@ -175,6 +175,7 @@ def generate_folded_ast(
     vy_ast.folding.fold(vyper_module_folded)
     validate_semantics(vyper_module_folded, interface_codes)
     vy_ast.expansion.expand_annotated_ast(vyper_module_folded)
+    set_data_positions(vyper_module_folded)
```

Comment:
It would be nice if the storage and memory layout generated during this phase were made available externally

Reply:
Yeah, makes sense as a compiler output.  But I think better done after the entirety of data positions are being calculated in that pass.